### DISCUSSION TITLE
Replace usage of yarn with ni?

### DISCUSSION THREAD

**Discussion by abevoelker** [rails/jsbundling-rails#127] [Ideas]
There are some past issues/discussion about replacing yarn (#68, #3, #6, maybe more) with npm or pnpm (I myself just use npm).

[ni is a tool](https://github.com/antfu/ni) that uses whatever JavaScript runtime the user has available at the time and supports npm, yarn, pnpm and bun. I wonder if it would be a good fit for JS Bundling since then it can just punt to whatever the user is using in their Rails project. The downside is it would be an additional dependency, however JS Bundling already requires yarn for people who don't use it.

It would replace these calls like this:

```
$ yarn install
$ yarn add rollup @rollup/plugin-node-resolve
$ yarn build --watch
```

to these, respectively:

```
$ ni
$ ni rollup @rollup/plugin-node-resolve
$ nr build --watch
```

**Comment by sandstrom** [comment]
Good idea, that or just replacing it with `npm` which is still the de-facto standard node package manager (it has it's flaws, but it works well enough and everybody knows how to use it).

**Comment by abevoelker** [comment]
Yeah that might be more straightforward. Presumably there’s a reason yarn
is being preferred by many Rails projects (e.g. the Turbo install task also
uses it) but I don’t know why. It’s definitely a fly in the ointment though
for those of us not using it
